pkg/npm/api/lib/BigIntOrderedMap.ts

```diff
@@ -1,227 +1,63 @@
-import { BigInteger } from "big-integer";
+import _ from 'lodash';
 import { immerable } from 'immer';
-
-interface NonemptyNode<V> {
-  n: [BigInteger, V];
-  l: MapNode<V>;
-  r: MapNode<V>;
+import bigInt, { BigInteger } from "big-integer";
+
+function sortBigInt(a: BigInteger, b: BigInteger) {
+  if (a.lt(b)) {
+    return 1;
+  } else if (a.eq(b)) {
+    return 0;
+  } else {
+    return -1;
+  }
 }
-
-type MapNode<V> = NonemptyNode<V> | null;
-
-/**
- *  An implementation of ordered maps for JS
- *  Plagiarised wholesale from sys/zuse
- */
 export default class BigIntOrderedMap<V> implements Iterable<[BigInteger, V]> {
-  private root: MapNode<V> = null;
+  private root: Record<string, V> = {}
+  private cachedIter: [BigInteger, V][] | null = null;
   [immerable] = true;
-  size: number = 0;
 
-  constructor(initial: [BigInteger, V][] = []) {
-    initial.forEach(([key, val]) => {
+  constructor(items: [BigInteger, V][] = []) {
+    _.forEach(items, ([key, val]) => {
```

Comment:
Couldn't you just do `items.forEach` and lose the lodash dependency?

Reply:
That would be nice.